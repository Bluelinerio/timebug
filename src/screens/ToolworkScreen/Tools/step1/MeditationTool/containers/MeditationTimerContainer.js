import { connect }           from 'react-redux'
import { mapProps, compose } from 'recompose'
import MeditationTimer       from '../components/MeditationTimer'

const mapDispatchToProps = (dispatch: any) => ({
  unlockAchievement: () => dispatch({ type: 'TODO_ACHIEVEMENT' }),
})

type Props = {
  onTimerFinish: () => any,
  daysInRowCount: number,
  disableTimer: boolean,
}

const merge = (props: Props) => {
  const { daysInRowCount, onTimerFinish, disableTimer } = props
  const multiplier = Math.min(Math.floor(daysInRowCount / 5) + 1, 12)
  const total = 300 * multiplier

  const onMeditationFinish = () => {
    onTimerFinish()
  }

  return {
    total,
    onMeditationFinish,
    disableTimer,
  }
}

export default compose(connect(null, mapDispatchToProps), mapProps(merge))(
  MeditationTimer
)
