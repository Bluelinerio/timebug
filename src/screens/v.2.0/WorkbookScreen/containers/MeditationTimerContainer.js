import { compose, mapProps }      from 'recompose'
import MeditationTimer, { Props } from '../components/MeditationTimer'
import { videoStatus as status }  from '../../../../HOC/AudioVideoComponentHOC'
import AudioVideoHOC              from '../../../../HOC/AudioVideoComponentHOC'

export type StepContentButtonContainerProps = {
  audio: string,
  color: string,
  videoStatus: string,
  handle: () => any,
}

const mergeProps = (ownProps: StepContentButtonContainerProps): Props => {
  const { color, videoStatus, handle, currentPosition, totalLength } = ownProps

  const icons = {
    [status.READY]: 'ios-play',
    [status.PLAYING]: 'ios-pause',
    [status.PAUSED]: 'ios-play',
    [status.FINISHED]: 'ios-play',
    [status.FAIL]: 'ios-close',
    [status.PENDING]: 'ios-time',
  }

  const icon = icons[videoStatus]

  return {
    color,
    icon,
    handle,
    currentPosition,
    totalLength,
    isPending: videoStatus === status.PENDING,
  }
}

export default compose(AudioVideoHOC, mapProps(mergeProps))(MeditationTimer)
