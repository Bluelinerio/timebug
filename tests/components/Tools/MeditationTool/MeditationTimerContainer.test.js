import React from 'react'
import shallowWithStore from '../../../shallowWithStore'
import { createMockStore } from 'redux-test-utils'
import MeditationTimerContainer from '../../../../src/screens/ToolworkScreen/Tools/step1/MeditationTool/containers/MeditationTimerContainer'

describe('Tests related to the meditation timer container', function() {
  const props = {
    daysInRowCount: 2,
    onTimerFinish: jest.fn(() => null),
    disableTimer: false,
    meditationData: {},
  }

  const state = {}

  it('should render', function() {
    const store = createMockStore(state)
    const component = shallowWithStore(
      <MeditationTimerContainer {...props} />,
      store
    )
    expect(component).toMatchSnapshot()
  })
})
