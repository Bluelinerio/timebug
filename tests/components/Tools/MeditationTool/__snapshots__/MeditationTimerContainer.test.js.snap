// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests related to the meditation timer container should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(mapProps(MeditationTimer))
    daysInRowCount={2}
    disableTimer={false}
    meditationData={Object {}}
    onTimerFinish={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "daysInRowCount": 2,
      "disableTimer": false,
      "meditationData": Object {},
      "onTimerFinish": [MockFunction],
      "unlockAchievement": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "daysInRowCount": 2,
        "disableTimer": false,
        "meditationData": Object {},
        "onTimerFinish": [MockFunction],
        "unlockAchievement": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;
